src/test/java/seedu/mtracker/console/AddForexParserTest.java

```diff
@@ -0,0 +1,125 @@
+package seedu.mtracker.console;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class AddForexParserTest {
+
+    public static final int PARAMETER_SIZE = 7;
+    private static final String STR_SEP = "%1$s";
+
+    public static final String[] EXPECTED_PARAMS_NO_REMARKS = {
+        "TTTXXX",
+        "1.11",
+        "positive",
+        "1.15",
+        "1.30",
+        "15 Oct",
+        ""
+    };
+
+    public static final String[] EXPECTED_PARAMS_WITH_REMARKS = {
+        "TTTXXX",
+        "0.81",
+        "negative",
+        "0.79",
+        "0.70",
+        "20 Oct",
+        "fooRemarks"
+    };
+
+    public static final String USER_INPUT_NO_REMARKS = "TTTXXX"
+            + STR_SEP + "1.11"
+            + STR_SEP + "positive"
+            + STR_SEP + "1.15"
+            + STR_SEP + "1.30"
+            + STR_SEP + "15 Oct"
+            + STR_SEP + " ";
+
+    public static final String USER_INPUT_WITH_REMARKS = "TTTXXX"
+            + STR_SEP + "0.81"
+            + STR_SEP + "negative"
+            + STR_SEP + "0.79"
+            + STR_SEP + "0.70"
+            + STR_SEP + "20 Oct"
+            + STR_SEP + "fooRemarks";
+
+    public static final String USER_INPUT_TRY_INVALID_NAME = STR_SEP.repeat(2) + "TTXX"
+            + STR_SEP + "TTTXXX"
+            + STR_SEP + "1.11"
+            + STR_SEP + "positive"
+            + STR_SEP + "1.15"
+            + STR_SEP + "1.30"
+            + STR_SEP + "15 Oct"
+            + STR_SEP + " ";
+
+    public static final String USER_INPUT_TRY_INVALID_PRICE = STR_SEP + "TTTXXX"
+            + STR_SEP + "lol"
+            + STR_SEP + "0.81"
+            + STR_SEP + "negative"
+            + STR_SEP + "0.79"
+            + STR_SEP + "0.70"
+            + STR_SEP + "20 Oct"
+            + STR_SEP + "fooRemarks";
+
+    public static final String USER_INPUT_TRY_INVALID_SENTIMENT = STR_SEP + "TTTXXX"
```

Comment:
The junit tests currently doesnt test for invalid entry price and exit price given that you said those parameters are compulsory. Create a new issue for this and don't forget to include the validations for those params

Reply:
Ok